weio/fast_output_file.py

```diff
@@ -481,125 +753,78 @@ def freadNumpy(fid, n, dtype):
         fread      = freadStruct
         freadLarge = freadNumpy
     elif method=='optim':
-        # Decide on method on the fly
-        #MB = getFileSizeMB(filename)
         use_buffer = False
         fread      = freadStruct
         freadLarge = freadNumpy
     else:
         raise NotImplementedError
 
     def freadRowOrderTableBuffered(fid, n, type_in, nCols, nOff=0, type_out='float64'):
-        """ 
+        """
         Reads of row-ordered table from a binary file.
 
-        Read `n` data of type `type_in`, assumed to be a row ordered table of `nCols` columns.
-        Memory usage is optimized by allocating the data only once.
-        Buffered reading is done for improved performances (in particular for 32bit python)
-
-        `nOff` allows for additional column space at the begining of the storage table.
-        Typically, `nOff=1`, provides a column at the beginning to store the time vector.
-
-        @author E.Branlard, NREL
-
+        `nOff` allows for additional column space at the beginning of the array
+            for instance for time
         """
-        fmt, nbytes = StructDict[type_in][:2]
-        nLines          = int(n/nCols)
-        GoodBufferSize  = 4096*40
-        nLinesPerBuffer = int(GoodBufferSize/nCols)
-        BufferSize      = nCols * nLinesPerBuffer
-        nBuffer         = int(n/BufferSize)
-        # Allocation of data
-        data = np.zeros((nLines,nCols+nOff), dtype = type_out)
-        # Reading
+        ## Allocation of the output
+        if type_out=='float64':
+            data = np.zeros((int(n/nCols), nCols+nOff), dtype=np.float64)
+        else:
+            raise NotImplementedError('Unsupported type {}'.format(type_out))
+        ## Reading data from file
         try:
             nIntRead   = 0
-            nLinesRead = 0
-            while nIntRead<n:
-                nIntToRead = min(n-nIntRead, BufferSize)
-                nLinesToRead = int(nIntToRead/nCols)
-                Buffer = freadLarge(fid, nIntToRead, type_in)
-                Buffer = Buffer.reshape(-1,nCols)
-                data[ nLinesRead:(nLinesRead+nLinesToRead),  nOff:(nOff+nCols)  ] = Buffer
-                nLinesRead = nLinesRead + nLinesToRead
+            iChunk = 0
+            nChunkSize = min(n, 1024000)  # twice a 8*Mb chunk
+            while nIntRead < n:
+                nIntToRead = min(n - nIntRead, nChunkSize)
+                iRow       = int((nIntRead/nCols))
+                nRows      = int(nIntToRead/nCols)
+                Buffer     = freadLarge(fid, nIntToRead, type_in)
+                data[iRow:iRow+nRows, nOff:] = np.array(Buffer).reshape(-1, nCols)

```

Comment:
above, AI has done unnecessary changes. I would keep the original code and variables names for safety